project_merge/wizard/project_task_merge.py

```diff
@@ -0,0 +1,94 @@
+from odoo import _, fields, models
+
+
+class ProjectTaskMerge(models.TransientModel):
+    _name = "project.task.merge"
+    _description = "Project Task Merge"
+
+    task_ids = fields.Many2many("project.task", string="Tasks to Merge", required=True)
+    user_ids = fields.Many2many("res.users", string="Assignees")
+    create_new_task = fields.Boolean("Create a new task")
+    dst_task_name = fields.Char("New task name")
+    dst_project_id = fields.Many2one("project.project", string="Destination Project")
+    dst_task_id = fields.Many2one("project.task", string="Merge into an existing task")
+
+    def merge_tasks(self):
+        tag_ids = self.task_ids.mapped("tag_ids").ids
+        attachment_ids = self.task_ids.mapped("attachment_ids").ids
+        values = {
+            "description": self.merge_description(),
+            "tag_ids": [(4, tag_id) for tag_id in tag_ids],
+            "attachment_ids": [(4, attachment_id) for attachment_id in attachment_ids],
+            "user_ids": self.user_ids.ids,
+        }
+        if self.create_new_task:
+            partner_ids = self.task_ids.mapped("partner_id")
+            priorities = self.task_ids.mapped("priority")
+            values.update(
+                {
+                    "name": self.dst_task_name,
+                    "project_id": self.dst_project_id.id,
+                    "partner_id": len(set(partner_ids)) == 1
+                    and partner_ids[0].id
+                    or False,
+                    "priority": len(set(priorities)) == 1 and priorities[0] or False,
+                }
+            )
+            self.dst_task_id = self.env["project.task"].create(values)
+        else:
+            self.dst_task_id.write(values)
+        merged_tasks = self.task_ids - self.dst_task_id
+        self._merge_followers(merged_tasks)
+        for task in merged_tasks:
+            self._add_message("to", self.dst_task_id.name, task)
+        task_names = ", ".join(merged_tasks.mapped("name"))
+        self._add_message("from", task_names, self.dst_task_id)
+        merged_tasks.write({"active": False})
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "project.task",
+            "views": [[False, "form"]],
+            "res_id": self.dst_task_id.id,
+        }
+
+    def merge_description(self):
+        return "<br/>".join(
+            self.task_ids.mapped(
+                lambda task: "Description from task <b>%s</b>:<br/>%s"
+                % (task.name, task.description or "No description")
```

Comment:
Functional suggestion: Don't add the description and "Description from task" if the description is empty

![image](https://github.com/user-attachments/assets/d7634a8c-30cd-4f2e-ac78-7b712fb857a1)

Reply:
Updated the changes